toolkit/tools/imagegen/installutils/installutils.go

```diff
@@ -939,6 +939,52 @@ func InstallGrubEnv(installRoot string) (err error) {
 	return
 }
 
+// CopyAndChangeModeGrubDefaults copies a file from src to dst, creating directories with the given 
+// access rights for the destination if needed. dst is assumed to be a file and not a directory.
+// Will change the permissions to the given value.
+// In the case of handling grub defaults, we must additionally preserve the state of dst, appending it
+// to the src template before stomping. This is to preserve any configurations that might have been
+// installed by packages before the tools install our template. 
+// - src is the Mariner-tools-defined template file
+// - dst holds the root partition
+func CopyAndChangeModeGrubDefaults(src, dst string, dirmode os.FileMode, filemode os.FileMode) (err error) {
+	const (
+		tmpFile = "/installer/grub2/grub.tmp"
```

Comment:
We should be able to just keep this in a buffer (`srcContent`), there won't be a lot of text in the file.
Can't we just read the contents of the file, stop the template over top of it, and write the lines back onto the end?

Reply:
Fair enough, will make that change.